go/vt/sqlparser/ast.go

```diff
@@ -2220,7 +2220,14 @@ func (ct *ColumnType) Format(buf *TrackedBuffer) {
 		opts = append(opts, keywordStrings[NOT], keywordStrings[NULL])
 	}
 	if ct.Default != nil {
-		opts = append(opts, keywordStrings[DEFAULT], String(ct.Default))
+		opts = append(opts, keywordStrings[DEFAULT])
+		_, isLiteral := ct.Default.(*Literal)
+		_, isNullVal := ct.Default.(*NullVal)
+		if isLiteral || isNullVal {
+			opts = append(opts, String(ct.Default))
+		} else {
+			opts = append(opts, "("+String(ct.Default)+")")
+		}
```

Comment:
Why don't we need to check if  `_, isFuncExpr := ct.Default.(*FuncExpr); isFuncExpr == true` ?

Reply:
Because stuff like this is also possible 🤯  -
```sql
MySQL root@(none):test> create table function_default3 (x bool DEFAULT (true AND false));
Query OK, 0 rows affected
Time: 0.008s
MySQL root@(none):test> insert into function_default3 values ();
Query OK, 1 row affected
Time: 0.002s
MySQL root@(none):test> select * from function_default3;
+---+
| x |
+---+
| 0 |
+---+
1 row in set
Time: 0.007s
```

Reply:
Added a parse test for this case too

Reply:
yeah, we keep talking about function calls here, but the syntax rules say "arbitrary expression".

Reply:
makes sense, thank you